drivers/serial/uart_mcux_lpuart.c

```diff
@@ -1017,8 +1021,31 @@ static int mcux_lpuart_configure_init(const struct device *dev, const struct uar
 		/* Set the LPUART into loopback mode */
 		config->base->CTRL |= LPUART_CTRL_LOOPS_MASK;
 		config->base->CTRL &= ~LPUART_CTRL_RSRC_MASK;
+	} else if (config->single_wire) {
+		/* Enable the single wire / half-duplex mode, only possible when
+		 * loopback is disabled We need a critical section and flush to prevent
+		 * the UART firing an interrupt during mode switch
+		 */
+		unsigned int key = irq_lock();
+
+		sys_cache_data_flush_all();
```

Comment:
not understanding from the comment why the flush is needed

Reply:
It's to force a seperate bus transaction using a barrier (I couldn't find a clean macro for dsb) since the bus can pipeline it with the previous one.

Reply:
the barriers API doesn't work for you ? https://docs.zephyrproject.org/latest/hardware/barriers/index.html
is total data cache flush necessary then or not?

Reply:
Okay found the culprit the LPUART_Deinit hanged on waiting on TX when being reinitialized when single-wire is enabled. 
I've added a mask deasserting `LPUART_CTRL_RSRC_MASK` when calling `mcux_lpuart_configure`.

Now the flush isn't needed.